tests/phpunit/CRM/Contribute/Form/Contribution/ConfirmTest.php

```diff
@@ -922,4 +922,95 @@ public function testSubmitMultiIntervalMembershipContributionPage(bool $isQuickC
     $this->assertEquals(2, $contributionRecur['frequency_interval']);
   }
 
+  public function testSubmitInHonorOfContributionPage(): void {
+    $tributeProfile = $this->callAPISuccess('UFGroup', 'create', [
+      'group_type' => [
+        'Individual',
+        'Contact',
+      ],
+      'title' => 'Tribute',
+      'frontend_title' => 'Tribute',
+    ]);
+    $fields = [
+      'First Name' => 'first_name',
+      'Last Name' => 'last_name',
+      'Street Address (Home)' => 'street_address',
+      'City (Home)' => 'city',
+      'Postal Code (Home)' => 'postal_code',
+      'Country (Home)' => 'country',
+      'State (Home)' => 'state_province',
+    ];
+    foreach ($fields as $label => $fieldName) {
+      $field_type = 'Individual';
+      if ($fieldName !== 'first_name' && $fieldName !== 'last_name') {
+        $field_type = 'Contact';
+      }
+      $params = [
+        'uf_group_id' => $tributeProfile['id'],
+        'field_name' => $fieldName,
+        'label' => $label,
+        'field_type' => $field_type,
+      ];
+      if ($field_type === 'Contact') {
+        $params['location_type_id'] = 1;
+      }
+      $this->callAPISuccess('UFField', 'create', $params);
+    }
+    $this->contributionPageWithPriceSetCreate();
+    $this->callAPISuccess('UFJoin', 'create', [
+      'is_active' => 1,
+      'module' => 'CiviEvent',
+      'entity_table' => 'civicrm_event',
+      'entity_id' => $this->getContributionPageID(),
+      'weight' => 1,
+      'uf_group_id' => 1,
+    ]);
+    $this->callAPISuccess('UFJoin', 'create', [
+      'is_active' => 1,
+      'module' => 'soft_credit',
+      'entity_table' => 'civicrm_contribution_page',
+      'entity_id' => 1,
+      'uf_group_id' => $tributeProfile['id'],
+      'module_data' => [
+        'soft_credit' => [
+         'soft_credit_types' =>
+          [
+            0 => '1',
+            1 => '2',
+          ],
+          'default' => [
+            'honor_block_title' => 'SL Test',
+            'honor_block_text' => '',
+          ],
+        ],
+      ],
+    ]);
+    $processor = \Civi\Payment\System::singleton()->getById($this->ids['PaymentProcessor']['dummy']);
+    $processor->setDoDirectPaymentResult(['payment_status_id' => 1, 'fee_amount' => .72]);
+    $this->submitOnlineContributionForm([
+      'priceSetId' => $this->getPriceSetID('ContributionPage'),
+      'price_' . $this->ids['PriceField']['radio_field'] => $this->ids['PriceFieldValue']['10_dollars'],
+      'id' => $this->getContributionPageID(),
+      'soft_credit_type_id' => 2,
+      'honor' => [
+        'first_name' => 'James',
+        'last_name' => 'Bond',
+        'street_address-1' => 'Vaxhaul Cross',
+        'city-1' => 'London',
+        'postal_code-1' => 'M4K1J1',
+        'country-1' => 1039,
+        'state_province-1' => 1068,
```

Comment:
I assume the unit test fail is jenkins pointing out that this other James Bond that lives in London, Ontario, Canada appears to have a Toronto postal code! Crazy!